app/queries/admin/users_query.rb

```diff
@@ -31,6 +46,22 @@ def self.search_relation(relation, search)
       relation.where(QUERY_CLAUSE, search: "%#{search.strip}%")
     end
 
+    def self.filter_joining_date(relation:, joining_start:, joining_end:, date_format:)
+      ui_formats_to_parse_format = {
```

Comment:
It was a bit challenging deciding how to pass the dates to the query, since the form inputs can either be in the format "DD/MM/YYYY" or "MM/DD/YYYY", and naturally we need to know which way around we're expecting them 🙃 

In the end, I've added a hidden input to the form which notes which date format is being used, and here we're pulling out what we then need for `strptime` to parse those strings.

Maybe there is a more elegant way to handle this?

I did consider using Javascript to maintain separate hidden inputs for `joining_start/end`, making the value of those inputs a more universally understood ISO string, but it felt like overkill to maintain separate inputs, and to rely on JS for making sure they stay in step with the UI

Reply:
Great rationalization.  I'm wondering if we want the above Hash declaration to be a constant?  Are we using either of these formats elsewhere in which the keys would be useful?

Reply:
Not at the moment, but yeah I pondered the same thing. I guess at some point in the future we will end up using the date range picker elsewhere, at which time we might find ourselves needing this info